server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestResolveIndexAction.java

```diff
@@ -68,7 +69,9 @@ protected BaseRestHandler.RestChannelConsumer prepareRequest(RestRequest request
                 ? null
                 : Arrays.stream(modeParam.split(","))
                     .map(IndexMode::fromString)
-                    .collect(() -> EnumSet.noneOf(IndexMode.class), EnumSet::add, EnumSet::addAll)
+                    .collect(() -> EnumSet.noneOf(IndexMode.class), EnumSet::add, EnumSet::addAll),
+            false,
+            crossProjectMode
```

Comment:
This will likely be on IndicesOptions instead of on the request instance instead.